plugins/mac-capture/mac-sck-common.m

```diff
@@ -0,0 +1,339 @@
+#include "mac-sck-common.h"
+
+bool is_screen_capture_available(void)
+{
+    if (@available(macOS 12.5, *)) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+#pragma mark - ScreenCaptureDelegate
+
+@implementation ScreenCaptureDelegate
+
+- (void)stream:(SCStream *)stream didOutputSampleBuffer:(CMSampleBufferRef)sampleBuffer ofType:(SCStreamOutputType)type
+{
+    if (self.sc != NULL) {
+        if (type == SCStreamOutputTypeScreen && !self.sc->audio_only) {
+            screen_stream_video_update(self.sc, sampleBuffer);
+        }
+#if __MAC_OS_X_VERSION_MAX_ALLOWED >= 130000
+        else if (@available(macOS 13.0, *)) {
+            if (type == SCStreamOutputTypeAudio) {
+                screen_stream_audio_update(self.sc, sampleBuffer);
+            }
+        }
+#endif
+    }
+}
+
+- (void)stream:(SCStream *)stream didStopWithError:(NSError *)error
+{
+    NSString *errorMessage;
```

Comment:
`errorMessage` is created, but never used. Using `error.domain.UTF8String` doesn't seem to make much sense below (because it doesn't give you an actual error description, that'd be `error.localizedDescription`.

Reply:
Hmm, looks like this came from https://github.com/obsproject/obs-studio/pull/9065/. I just pushed a change with ``error.domain.UTF8String`` changed to ``errorMessage.UTF8String``, since I assume that was Sebastian's intention. I'll ping @gxalpha just in case as well.

Reply:
> I just pushed a change with error.domain.UTF8String changed to errorMessage.UTF8String, since I assume that was Sebastian's intention

Yes, that was an embarrassing mistake in the original which resulted from a test that then somehow made its way in the implementation. See https://github.com/obsproject/obs-studio/pull/9107 where I included a fix for that.

Reply:
Alright, will probably merge this later tonight then.